lewis/adapters/stream.py

```diff
@@ -78,13 +78,29 @@ def __init__(self, host, port, target):
         self.bind((host, port))
         self.listen(5)
 
+        self._accepted_connections = []
+
     def handle_accept(self):
         pair = self.accept()
         if pair is not None:
             sock, addr = pair
             print("Client connect from %s" % repr(addr))
             StreamHandler(sock, self.target)
 
+            self._accepted_connections.append(sock)
```

Comment:
I tried manually by disconnecting and I couldn't see any adverse behavior (exceptions etc), so I assume calling `.close` multiple times doesn't matter for the socket. What I'm not sure about are the implications of keeping the object around.

We could also try storing the `StreamHandler` and calling `close` on that instead. That would not change the fact that we'd keep the handler around for longer than necessary.